### ISSUE TITLE
Error when uploading file over websocket connection

### ISSUE THREAD

**Issue by ParanoidVoxel** [Xpra-org/xpra#3414]
**Describe the bug**
When dragging and dropping a file in the xpra web-ui, the filetransfer fails immediately on the server side with the following error:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/xpra/server/server_base.py", line 906, in process_packet
    call_handler()
  File "/usr/lib/python3/dist-packages/xpra/server/server_base.py", line 896, in call_handler
    handler(proto, packet)
  File "/usr/lib/python3/dist-packages/xpra/server/mixins/fileprint_server.py", line 250, in _process_send_file
    ss._process_send_file(packet)
  File "/usr/lib/python3/dist-packages/xpra/net/file_transfer.py", line 496, in _process_send_file
    os.write(fd, file_data)
```
**To Reproduce**
1. Start xpra with: `xpra start-desktop :80 --start=/opt/workspace/run_emwm.sh --webcam=no --exit-with-children=no --no-daemon --speaker=on --video-encoders=x264 --speaker-codec=wav,opus+mka --min-speed=75` (`run_emwm.sh` is a script that just launches my window manager)
2. Connect to the xpra web-ui running on the same server
3. Drag and drop file over browser window

**System Information (please complete the following information):**
 - Server OS: Ubuntu Bullseye (Docker)
 - Client OS: Web-UI running in the same container, accessed with chrome and firefox, both on w10 (same issue with both)
 - Xpra Server Version: v4.3.1-r0
 - Xpra Web-Client Version: 4.5.2

**Additional context**
This issue doesn't affect the "full" xpra client connecting with TCP. Could be an issue with only web-socket clients?

However, after implementing this dirty quickfix in the xrpa server, the issue is fixed in the web client, and file upload from the full client (using tcp) isn't broken by this change.

```diff
--- /usr/lib/python3/dist-packages/xpra/net/file_transfer.py    2022-01-07 17:08:03.779462400 +0000
+++ ./file_transfer.py  2022-01-07 17:07:55.389462400 +0000
@@ -491,6 +491,7 @@
         check_digest("sha256", hashlib.sha256)
         check_digest("sha1", hashlib.sha1)
         check_digest("md5", hashlib.md5)
+        file_data = file_data.encode("latin")
         try:
             os.write(fd, file_data)
         finally:
```

I didn't want to submit the pull-request because of the "quick-and-dirty" nature of my fix, and since I wasn't able to find where the `file_data` was being decoded with latin1 in the first place (probably where the websocket traffic is being handled), figured you guys would know better. 

I get decoding the other fields in the packet (like mimetype, filesize etc) but not the raw byte-data of the file.

Let me know if there's more info that I can provide, thanks in advance!

**Comment by totaam** [comment]
The type of `file_data` will depend on the packet encoder.
We want `bytes` because we really don't want to have to do anything with the data, just save it as it is. Unfortunately, some packet encoders may give us `str` instead. (ie: `rencode`)
At first glance, the html5 client looks wrong and uses a `Uint8ToString` conversion:
https://github.com/Xpra-org/xpra-html5/blob/311a1768ce052575a6225951c3bce7d8b0a26413/html5/index.html#L1103-L1111
When it shouldn't need to do this for the new `rencodeplus` packet encoder.
For older packet encoders, we may still need the server side fix.

**Comment by totaam** [comment]
The commit above is the same workaround you had implemented, only safer.
The real fixes belong in the html5 client: abe0a9690bc9b3aab8f4969bd7afff522149ae90 then also uncovered 99f4369ab0ece970b8c2d4699592fedcaa2b0c30.

I recommend the html5 client fixes, they're just more correct.
(rather than doing an expensive and unnecessary roundtrip via strings at both ends)

**Comment by ParanoidVoxel** [comment]
Awesome work, thank you!

Yeah the html5-client fixes is what I was trying to get at but I couldn't figure out where I could implement it. I guess we can consider this issue closed then? :)
